src/components/organisms/AdminDashboard/AdminDashboard.tsx

```diff
@@ -29,10 +30,13 @@ import "./AdminDashboard.scss";
 export const AdminDashboard: React.FC = () => {
   const { ownedVenues, isLoading } = useOwnedVenues({});
 
-  const { worldId } = useWorldEditParams();
+  const { worldSlug } = useWorldEditParams();
 
-  const venues = worldId
-    ? ownedVenues.filter((venue) => venue.worldId === worldId)
+  const { world, isLoaded } = useWorldBySlug(worldSlug);
```

Comment:
Might be easier to use if a single `useWorld()` hook exists that can inspect its options

```ts
const { world, isLoaded } = useWorld({ worldSlug });
```
```ts
const { world, isLoaded } = useWorld({ worldId });
```

Reply:
`reactfire` is using a lot of hooks and it will be difficult to put conditions in a single hook and make it work fast, I will most likely have to fetch both worlds by Id and slug and then use the one with a result, because we can't use hooks after conditions. It won't be very good for the performance of the platform.

Reply:
They also return different data, since one of the hooks is searching by the doc id and the other one with a `where` operator

Reply:
`They also return different data` <- still the type `World`, right?

So, just to double check, you say something like the following will be slow:
```ts

  const queryBySlug = worldsRef
    .where("slug", "==", worldSlug ?? "")
    .withConverter(worldConverter);
  const { data: worlds, status: statusBySlug } = useFirestoreCollectionData<WithId<World>>(queryBySlug, { initialData: undefined });

  const queryById = worldsRef.doc(worldId).withConverter(worldConverter);
  const { data: world, status: statusById } = useFirestoreDocData<WithId<World>>(queryById, { initialData: undefined });

  const isLoaded = (worldId ? statusById : statusBySlug) !== "loading";

  return {
    world: worldId ? world : worlds?.[0],
    isLoaded: isLoaded,
  };
```

I'm not sure I can make that claim without performance test, but If that's the case, I should probably make some small PR later on to make the hooks appear similar in name.

Reply:
Yes this is how that hook is supposed to look like but there is always going to be an unnecessary search. If I'm passing down an ID the logic shouldn't search by slug, but since we can't use conditions in before hooks, it always will. And it's a bad practice to trade performance for code quality or readability. The safer approach would be to use 2 separate hooks and don't risk this.